tests/pipeline/test_term.py

```diff
@@ -321,3 +325,31 @@ class SomeFactorNoDType(SomeFactor):
 
         with self.assertRaises(InvalidDType):
             SomeFactor(dtype=1)
+
+
+class SubDataSetTestCase(TestCase):
+    def test_subdataset(self):
+        some_dataset_map = {
+            column.name: column for column in SomeDataSet.columns
+        }
+        sub_dataset_map = {
+            column.name: column for column in SubDataSet.columns
+        }
+        self.assertEqual(
```

Comment:
I was confused by this assertion when reading.  I think it'd be clearer to assert that `SomeDataSet.columns` is equal to `SubDataSet.columns`, which I think it both equivalent and closer to the intent of what's being tested here.